### PR TITLE
Update ironic node resource object to work with node owners/lessees

### PR DESCRIPTION
**Pull request by tzumainn**
Ironic nodes have changed to have specific owner/lessee fields.
This patch updates the esi-leap code to match those changes.

### CODE REVIEW

File: esi_leap/resource_objects/ironic_node.py [hunk 1]
```diff
@@ -49,14 +49,12 @@ def get_contract_uuid(self):
...
 
     def get_node_config(self):
         node = get_ironic_client().node.get(self._uuid)
         config = node.properties
         config.pop('contract_uuid', None)
-        config.pop('project_id', None)
-        config.pop('project_owner_id', None)
         return config
 
     def set_contract(self, contract):
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply esi_leap/resource_objects/ironic_node.py:60]: Not quite! It should be the lessee.

[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/resource_objects/ironic_node.py:60]: Got it.

File: esi_leap/resource_objects/ironic_node.py [hunk 2]
```diff
@@ -49,14 +49,12 @@ def get_contract_uuid(self):
...
         node = get_ironic_client().node.get(self._uuid)
-        return node.properties.get('project_id', None)
+        return node.lessee
 
     def get_node_config(self):
         node = get_ironic_client().node.get(self._uuid)
         config = node.properties
         config.pop('contract_uuid', None)
-        config.pop('project_id', None)
-        config.pop('project_owner_id', None)
```
Comment (cudkowicz) [inline esi_leap/resource_objects/ironic_node.py:59]: Why get rid of these lines?

Comment (tzumainn) [reply esi_leap/resource_objects/ironic_node.py:59]: Ah, this was the old way for an Ironic node to know who owned/leased it - a hackish way not supported upstream. This is now the official Ironic way to handle this.

**Pull request merged**